shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs

```diff
@@ -800,9 +802,10 @@ verifiedWits ::
     DSignable crypto (Hash crypto (TxBody crypto))
   ) =>
   Tx crypto ->
-  Bool
-verifiedWits (Tx txbody wits _ _) =
-  all (verifyWitVKey $ hashWithSerialiser toCBOR txbody) wits
+  [VKey 'Witness crypto]
```

Comment:
This is very confusing! The function `verifiedWits` returns the list of witnesses that _didn't_ verify?

In such a case, I think it's better to use `Either [VKey 'Witness crypto] ()` (or at least a heavy comment flagging the unexpected behaviour!)

Reply:
yes, thank you, that's way better